R/activate.R

```diff
@@ -156,22 +156,26 @@ renv_activate_prompt <- function(action, library, prompt, project) {
   if (!ask)
     return(FALSE)
 
-  fmt <- lines(
-    "",
-    "This project has not yet been activated.",
-    "Activating this project will ensure the project library is used when %s() is called.",
-    "Please see `?renv::activate` for more details.",
-    ""
-  )
-  writef(fmt, action)
+  renv_activate_prompt_impl(action, project)
 
-  fmt <- "Would you like to activate this project before %s() is called?"
-  question <- sprintf(fmt, action)
-  response <- ask(question, default = TRUE)
-  if (!response)
-    return(FALSE)
 
-  activate(project = project)
-  return(TRUE)
+}
 
+renv_activate_prompt_impl <- function(action, project = NULL) {
+  title <- c(
+    sprintf("It looks like you've called renv::%s() in a non-renv project.", action),
+    "How would you like to proceed?"
+  )
+  choices <- c(
+    activate = "Activate the project and use the project library.",
+    continue = "Do not activate the project and use the current library paths.",
+    cancel = "Cancel and resolve the problem another way."
+  )
+
+  choice <- menu(choices, title, default = 2)
```

Comment:
Do we accept named defaults here, for when "choices" is a named character vector?

Reply:
Yup, and that's easier to reason about.